src/Storages/MergeTree/MergeTreeWhereOptimizer.h

```diff
@@ -104,6 +106,7 @@ class MergeTreeWhereOptimizer : private boost::noncopyable
     String first_primary_key_column;
     const StringSet table_columns;
     const Names queried_columns;
+    const Names primary_key_columns;
```

Comment:
It's better to save these name to `unordered_set` to avoid linear search for every check of column.